src/handle.jl

```diff
@@ -14,13 +14,43 @@
 abstract AbstractSundialsObject
 
 immutable CVODEMem <: AbstractSundialsObject end
+type CVODEMemContainer <: AbstractSundialsObject
+    ptr::Ptr{CVODEMem}
+    function CVODEMemContainer(lmm, iter)
+        ptr = new(CVodeCreate(lmm,iter))
+        finalizer(ptr,cvode_finalizer)
+        ptr
+    end
+end
 typealias CVODEMemPtr Ptr{CVODEMem}
+cvode_finalizer(mem::CVODEMemContainer) = mem.ptr == C_NULL || (CVodeFree(Ref{CVODEMemPtr}(mem.ptr)) ; mem.ptr = C_NULL)
+convert(::Type{CVODEMemPtr},mem::CVODEMemContainer) = mem.ptr
```

Comment:
It's here and other places: https://github.com/JuliaDiffEq/Sundials.jl/blob/master/src/cvode.jl

These files are autogenerated using Clang.jl so will be re-written every time the Sundials version is updated. So it's easiest just to use the conversion that it has to get the pointer.

I don't think the ptr check needs to happen every time a Sundials function is called, right? I'd think it would be enough to check on creation and in the finalizer.